### DISCUSSION TITLE
Will the Raspbian 11 (Bullseye) bundle work on Raspbian 12 (Bookworm)

### DISCUSSION THREAD

**Discussion by brianddk** [Motion-Project/motion#1709] [Q&A]
Unpacking the [pi_bullseye_motion_4.5.1-1_armhf.deb][1] I have the following broken dependencies:

1.  `libavcodec59` instead of  of `libavcodec58`
2. `libavdevice59` instead of `libavdevice58`
2. `libavformat59` instead of `libavformat58`
2. `libavutil57` instead of `libavutil56`
2. `libswscale6` instead of `libswscale5`

I'm fine with editing the `DEBIAN/control` file or running some of the `dpkg --force` options, but was curious, from a technical aspect, if the same binaries would likely work on both Bullseye and Bookworm?  But since the binaries in the Buster and Bullseye deps are different, then perhaps there is some statically linked library that would blowup if I tried one of them on Bookworm.

Thoughts?

[1]: https://github.com/Motion-Project/motion/releases/download/release-4.5.1/pi_bullseye_motion_4.5.1-1_armhf.deb

PS: I'll also try to build from src, but I'll have to use `libjpeg9-dev` instead of `libjpeg8-dev` which built clean at least.

**Comment by brianddk** [comment]
I went ahead and did [the following][1] just to be safe.

1. Built `motion` from source using `libjpeg9-dev` instead of `libjpeg8-dev`
2. Unpacked the [pi_bullseye_motion_4.5.1-1_armhf.deb][2] as root
3. Modified `DEBIAN/control`, `DEBIAN/md5sum` and replaced `usr/bin/motion` with source build (as root)
4. Repacked file as `modified.deb` and installed it.
5. Followed [setup instructions][3]

[1]: https://unix.stackexchange.com/questions/138188/easily-unpack-deb-edit-postinst-and-repack-deb
[2]: https://github.com/Motion-Project/motion/releases/download/release-4.5.1/pi_bullseye_motion_4.5.1-1_armhf.deb
[3]: https://pimylifeup.com/raspberry-pi-webcam-server/

Once I get the camera hooked up I'll report back on how it goes.

**Comment by tosiara** [comment]
`4.5.1` has been released before Bookworm, that's why the Release page is missing a build for it.

Can you give a try to this deb?

[bookworm_motion_4.5.1-1_armel.deb.zip](https://github.com/Motion-Project/motion/files/12567706/bookworm_motion_4.5.1-1_armel.deb.zip)

[bookworm_motion_4.5.1-1_armhf.deb.zip](https://github.com/Motion-Project/motion/files/12567755/bookworm_motion_4.5.1-1_armhf.deb.zip)

If it works fine for you, I will also update the Release page

**Reply by pcollinson** [reply]
I presume that this is the standard version included on the bookworm distro. I got it to work with my old RPi camera by:

a) apt install libcamera-v4l2
b) changing the command line in /lib/systemd/system/motion.service to read
ExecStart=libcamerify /usr/bin/motion

You might like to expand on the comments on https://motion-project.github.io/motion_config.html - this did give me the tip about libcamerify.

**Reply by pcollinson** [reply]
Hmm. Well I spoke too soon. I've always asked for an image that is 1024x576 from motion. My camera is a V2 Pi Camera. All previous releases have delivered an image that is a scaled down version of the full image (maybe) that the camera is sending. Using the new bookworm release, motion delivers a small viewport from the centre of image.

It seems that on bullseye, asking for the smaller image will scale it down, on bookworm it extracts a sized part of the image. Both systems appear to pick the YU12 palette, whatever that is:-) 

I couldn't find an easy way to get around this. So I've reverted the system to bullseye. Don't fix what ain't broke.

Also on bookworm, Motion doesn't seem to be able to set the frame rate - which may or may not be a problem.

The 'maybe' above is because maybe the full image is not delivered by the bullseye release, but if it does viewport the image it's certainly most of what it can see.
